### PR TITLE
[DEV-3832] Add Strapi Wrappers around Factories and Fixture

### PR DESCRIPTION
**Pull request by Sebastiano-Bertolin**
#### List of Changes

**Refactor product factories and tests; consolidate fixtures**
- Created a new factories file for product-related tests to improve organization and maintainability.
- Removed the old product factories file and migrated relevant functions to the new factories file.
- Introduced a new fixtures file for products to centralize test data and reduce duplication.
- Updated all test files to reference the new fixtures and factories structure.
- Adjusted solution list page and tutorial tests to align with the new data structure.
- Enhanced webinar tests with new factory functions for handling edge cases.

#### Types of changes

- [x] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: apps/nextjs-website/src/lib/__tests__/strapiEntityWrappers.ts [hunk 1]
```diff
@@ -12,9 +12,9 @@ export function wrapAsPaginatedRootEntity<T>(
     data: entities,
     meta: {
       pagination: {
-        page: 1,
+        page: 0,
         pageSize: entities.length,
-        pageCount: 1,
+        pageCount: 0,
         total: entities.length,
```
[the remark this answers is not in the document]
Comment (Sebastiano-Bertolin) [reply apps/nextjs-website/src/lib/__tests__/strapiEntityWrappers.ts]: Pagination data are not used by unit tests nor the webapp, those values are totally arbitrary

**Review by marcobottaro — APPROVED** [review]
